<javaelement>
	<package>
		java.sql
	</package>
	<type>
		<name>
			public class SQLRecoverableException  extends SQLException
		</name>
		<description>
			The subclass of SQLException thrown in situations where a previously failed operation might be able to succeed if the application performs some recovery steps and retries the entire transaction or in the case of a distributed transaction, the transaction branch. At a minimum, the recovery operation must include closing the current connection and getting a new connection.
		</description>
		<methods>
			<method>
				<name>
					public SQLRecoverableException()
				</name>
				<description>
					Constructs a SQLRecoverableException object. The reason, SQLState are initialized to null and the vendor code is initialized to 0. The cause is not initialized, and may subsequently be initialized by a call to the Throwable.initCause(java.lang.Throwable) method.
				</description>
				
			</method>
			<method>
				<name>
					public SQLRecoverableException(String reason)
				</name>
				<description>
					Constructs a SQLRecoverableException object with a given reason. The SQLState is initialized to null and the vender code is initialized to 0. The cause is not initialized, and may subsequently be initialized by a call to the Throwable.initCause(java.lang.Throwable) method.
				</description>
				
			</method>
			<method>
				<name>
					public SQLRecoverableException(String reason,                         String SQLState)
				</name>
				<description>
					Constructs a SQLRecoverableException object with a given reason and SQLState. The cause is not initialized, and may subsequently be initialized by a call to the Throwable.initCause(java.lang.Throwable) method. The vendor code is initialized to 0.
				</description>
				
			</method>
			<method>
				<name>
					public SQLRecoverableException(String reason,                         String SQLState,                         int vendorCode)
				</name>
				<description>
					Constructs a SQLRecoverableException object with a given reason, SQLState and vendorCode. The cause is not initialized, and may subsequently be initialized by a call to the Throwable.initCause(java.lang.Throwable) method.
				</description>
				
			</method>
			<method>
				<name>
					public SQLRecoverableException(Throwable cause)
				</name>
				<description>
					Constructs a SQLRecoverableException object with a given cause. The SQLState is initialized to null and the vendor code is initialized to 0. The reason is initialized to null if cause==null or to cause.toString() if cause!=null.
				</description>
				
			</method>
			<method>
				<name>
					public SQLRecoverableException(String reason,                         Throwable cause)
				</name>
				<description>
					Constructs a SQLRecoverableException object with a given reason and cause. The SQLState is initialized to null and the vendor code is initialized to 0.
				</description>
				
			</method>
			<method>
				<name>
					public SQLRecoverableException(String reason,                         String SQLState,                         Throwable cause)
				</name>
				<description>
					Constructs a SQLRecoverableException object with a given reason, SQLState and cause. The vendor code is initialized to 0.
				</description>
				
			</method>
			<method>
				<name>
					public SQLRecoverableException(String reason,                         String SQLState,                         int vendorCode,                         Throwable cause)
				</name>
				<description>
					Constructs a SQLRecoverableException object with a given reason, SQLState, vendorCode and cause.
				</description>
				
			</method>
			
		</methods>
		
	</type>
	
</javaelement>
